momepy/coins.py

```diff
@@ -0,0 +1,539 @@
+
+        self.split = outLine
+
+    def uniqueID(self):
+    #Loop through split lines, assign unique ID and
+    #store inside a list along with the connectivity dictionary
+        self.unique = dict(enumerate(self.split))
+
+    def getLinks(self):
+        global result
+        print("Finding adjacent segments...")
+
+        self.tempArray = np.array(self.tempArray, dtype=object)
+        
+        iterations = [n for n in range(0,len(self.unique))]
+        
+        pool = multiprocessing.Pool(multiprocessing.cpu_count())
+        constantParameterFunction = partial(_getLinksMultiprocessing, total=len(self.unique), tempArray=self.tempArray)
+        result = pool.map(constantParameterFunction, iterations)
+        pool.close()
+        pool.join()
+        iterations = None
+
+        for a in result:
+            n = a[0]
+            self.unique[n][2] = a[1]
+            self.unique[n][3] = a[2]
+            
+        print('>'*50 + ' [%d/%d] '%(len(self.unique),len(self.unique)) + '100%' + '\n', end='\r')
+            
+    def bestLink(self):
+        self.anglePairs = dict()
+        for edge in range(0,len(self.unique)):
+            p1AngleSet = []
+            p2AngleSet = []
+
+            """
+            Instead of computing the angle between the two segments twice, the method calculates
+            it once and stores in the dictionary for both the keys. So that it does not calculate
+            the second time because the key is already present in the dictionary.
+            """
+            for link1 in self.unique[edge][2]:
+                self.anglePairs["%d_%d" % (edge, link1)] = _angleBetweenTwoLines(self.unique[edge][0], self.unique[link1][0])
+                p1AngleSet.append(self.anglePairs["%d_%d" % (edge, link1)])
+                
+            for link2 in self.unique[edge][3]:
+                self.anglePairs["%d_%d" % (edge, link2)] = _angleBetweenTwoLines(self.unique[edge][0], self.unique[link2][0])
+                p2AngleSet.append(self.anglePairs["%d_%d" % (edge, link2)])
+
+            """
+            Among the adjacent segments deflection angle values, check for the maximum value
+            at both the ends. The segment with the maximum angle is stored in the attributes
+            to be cross-checked later for before finalising the segments at both the ends.
+            """
+            if len(p1AngleSet)!=0:
+                val1, idx1 = max((val, idx) for (idx, val) in enumerate(p1AngleSet))
+                self.unique[edge][4] = self.unique[edge][2][idx1], val1
+            else:
+                self.unique[edge][4] = 'DeadEnd'
+                
+            if len(p2AngleSet)!=0:
+                val2, idx2 = max((val, idx) for (idx, val) in enumerate(p2AngleSet))
+                self.unique[edge][5] = self.unique[edge][3][idx2], val2
+            else:
+                self.unique[edge][5] = 'DeadEnd'
+
+    def crossCheckLinks(self, angleThreshold):
+        global edge, bestP1, bestP2
```

Comment:
Same here. No need for global variables.

Reply:
Sounds good. From looking at the code, it seems like I don't need to make them `bestP1` or `bestP2` class variables. I am a but unsure about `edge`. However, it seems like if I just comment out this line, things are still working.